compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaTupleMemberSymbol.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.ballerina.compiler.api.impl.symbols;
+
+import io.ballerina.compiler.api.impl.SymbolFactory;
+import io.ballerina.compiler.api.symbols.AnnotationAttachmentSymbol;
+import io.ballerina.compiler.api.symbols.AnnotationSymbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.api.symbols.TupleMemberSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BAnnotationAttachmentSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BVarSymbol;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Represents a tuple member symbol.
+ *
+ * @since 2201.4.0
+ */
+public class BallerinaTupleMemberSymbol extends BallerinaSymbol implements TupleMemberSymbol {
+
+    private List<AnnotationSymbol> annots;
+    private List<AnnotationAttachmentSymbol> annotAttachments;
+    private final TypeSymbol type;
+
+    public BallerinaTupleMemberSymbol(CompilerContext context,
+                                      BVarSymbol symbol,
+                                      TypeSymbol type) {
+        super(null, SymbolKind.TUPLE_MEMBER, symbol, context);
+        this.type = type;
+    }
+
+    @Override
+    public Optional<String> getName() {
+        return Optional.empty();
+    }
+
+    public TypeSymbol getType() {
```

Comment:
Ditto.

```suggestion
    public TupleTypeSymbol typeDescriptor() {
```

Reply:
Renaming the `getType()` to `typeDescriptor()` is okay. But, why we should cast the TypeSymbol to TupleTypeSymbol ? we don't always get a tuple-type